src/interface/swarm.cpp

```diff
@@ -1087,11 +1090,11 @@ void Swarm::UnloadBuffers_() {
   auto &bdvar = vbswarm->bd_var_;
 
   if (total_received_particles_ > 0) {
-    ParArrayND<int> new_indices;
-    auto new_mask = AddEmptyParticles(total_received_particles_, new_indices);
+    auto newParticlesContext = AddEmptyParticles(total_received_particles_);
 
-    ParArrayND<int> neighbor_index("Neighbor index", total_received_particles_);
-    ParArrayND<int> buffer_index("Buffer index", total_received_particles_);
+    // TODO(BRR) remove these dynamic allocs
```

Comment:
As above, this is meant as a future to do and not part of this PR, correct? I'm just asking because thisPR is about dynamic allocation.

Reply:
I think this PR is the right place to remove this dynamic alloc so I did :+1: thanks for highlighting it